workflows/metatdenovo.nf

```diff
@@ -1,65 +1,167 @@
 /*
-========================================================================================
-    VALIDATE INPUTS
-========================================================================================
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    PRINT PARAMS SUMMARY
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-def summary_params = NfcoreSchema.paramsSummaryMap(workflow, params)
+include { paramsSummaryLog; paramsSummaryMap } from 'plugin/nf-validation'
 
-// Validate input parameters
-WorkflowMetatdenovo.initialise(params, log)
+def logo = NfcoreTemplate.logo(workflow, params.monochrome_logs)
+def citation = '\n' + WorkflowMain.citation(workflow) + '\n'
+def summary_params = paramsSummaryMap(workflow)
+
+// Validate parameters for orf_caller:
+ORF_CALLER_PRODIGAL     = 'prodigal'
+ORF_CALLER_PROKKA       = 'prokka'
+ORF_CALLER_TRANSDECODER = 'transdecoder'
+
+// Validate parameters for assembler:
+RNASPADES = 'rnaspades'
+MEGAHIT   = 'megahit'
+
+// validate parameters for eukulele database:
+EUKULELE_DB_PHYLODB     = 'phylodb'
+EUKULELE_DB_MMETSP      = 'mmetsp'
+EUKULELE_DB_EUKPROT     = 'eukprot'
+EUKULELE_DB_EUKZOO      = 'eukzoo'
+EUKULELE_DB_GTDB        = 'gtdb'
+
+def valid_params = [
+    orf_caller      : [ ORF_CALLER_PRODIGAL, ORF_CALLER_PROKKA, ORF_CALLER_TRANSDECODER ],
+    assembler       : [ RNASPADES, MEGAHIT ],
+    eukulele_db     : [ EUKULELE_DB_PHYLODB, EUKULELE_DB_MMETSP, EUKULELE_DB_EUKPROT, EUKULELE_DB_EUKZOO, EUKULELE_DB_GTDB ]
+]
```

Comment:
Hm, I think that can be done as mentioned before in the `nextflow_schema.json` with `enum`. The advantage is that the documentation on the website will contain the identical information. Easier to use and to maintain.